src/types.h

```diff
@@ -292,7 +293,11 @@ class SizedType {
   size_t GetNumElements() const
   {
     assert(IsArrayTy() || IsStringTy());
-    return IsStringTy() ? size_bits_ : size_bits_ / element_type_->size_bits_;
+    // For array's we can't just do size_bits_ / element_type.GetSize()
+    // because we might not know the size of the element type (it may be 0)
+    // if it's being resolved in a later AST pass e.g. for an imported type:
+    // `let $x: struct Foo[10];`
+    return IsStringTy() ? size_bits_ : num_elements_;
```

Comment:
Need to return bytes

```suggestion
    return IsStringTy() ? GetSize() : num_elements_;
```

Reply:
This was the previous logic. I'll change it in a follow up PR. I just changed the right side of the ternary.